src/poetry/core/packages/dependency.py

```diff
@@ -559,8 +553,7 @@ def create_from_pep_508(
                 name = m.group("name")
                 version = m.group("ver")
 
-            name = req.name or link.egg_fragment
```

Comment:
I am pretty sure this is just dead code -- more investigation and a possible removal of the field would be welcome.

Reply:
I reached the same conclusion - and removed it! are you asking me to do something more?

Reply:
oh, you mean the `egg_fragment`?  I decided that removing that might make it harder to get this MR merged, so left it for another day.